### PR TITLE
Add test for specialization constants via handler

### PR DESCRIPTION
**Pull request by vasilytric**
Provides tests for specialization constants via handler

Based on #140  (got common code (tests/specialization_constants/specialization_constants_common.h) from this PR)

Please review from commit: ef09455e3934f62dbbc29a8a2e4e21aac7764139

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

**Comment by bader** [comment]
@vasilytric, could you apply suggestions from @keryell, please?

**Comment by vasilytric** [comment]
> @vasilytric, could you apply suggestions from @keryell, please?

@bader Yes, but in Monday, next week.
This will be good time?

**Comment by bader** [comment]
Next Monday works for me.

File: tests/specialization_constants/specialization_constants_via_handler_common.h [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    {
+      const int case_num = 1;
+      result = T(value_helper<T>(0));
+      queue.submit([&](sycl::handler &cgh) {
+        cgh.set_specialization_constant<spec_const<T, case_num>>(ref);
+        result = cgh.get_specialization_constant<spec_const<T, case_num>>();
+      });
+    }
+    if (!check_equal_values(ref, result))
+      FAIL(log, "case 1 for " + type_name_string<T>::get(type_name));
```
[the remark this answers is not in the document]
Comment (kopylovanat) [reply tests/specialization_constants/specialization_constants_via_handler_common.h]: Changed to use case_num in error message

**Review by keryell — APPROVED** [review]
Thanks!

**Pull request merged**